Dump of current DB from server. (#143)
GermanZero-de/klimaschutzmonitor: README.md

This does not work when testing migrations, but there is a way: Use `read_fixture` in `cpmonitor/tests/migrations_test.py`.

## Manual tests with data from production

Occasionally, someone with access may provide a copy of the current production database. (See "Server Administration", below.)
This may be used as follows:

```sh
# Select the snapshot to use
SNAPSHOT_NAME=prod_database_<some date found in e2e_tests/database/>

# Remove previous data
rm db.sqlite3
rm -r cpmonitor/images/uploads

# Create DB
python manage.py migrate --settings=config.settings.local

> # Optionally migrate back to suitable version. See the corresponding README:

Reviewer: This line was very confusing for me until I read about the .README files further down, so let's go with something like this
```suggestion
# Optionally migrate back to suitable version (see the .README file corresponding to the snapshot you're about to load):
```

python manage.py migrate cpmonitor <some-earlier-migration> --settings=config.settings.local
python manage.py loaddata --settings=config.settings.local e2e_tests/database/${SNAPSHOT_NAME}.json
cp -r e2e_tests/database/${SNAPSHOT_NAME}_uploads cpmonitor/images/uploads
```

> If this is based on an older model version, migrations have to be applied and are tested:

Reviewer: ```suggestion
If snapshot you want to use is based on an older model version, migrations have to be applied and are tested:
```

## Server administration

### SSH access

Team members with their SSH keys installed on the server can add this to their `~/.ssh/config` file:

```.ssh/config
Host lzm
    HostName monitoring.localzero.net
    User monitoring
    IdentityFile ~/.ssh/<private_key_file>
```

This enables easy login:

```sh
ssh lzm
```

### Retrieve current DB from server

Replace your local DB with the current DB from the server with:

```sh
rm db.sqlite3
scp lzm:testing/db.sqlite3 .
rm -r cpmonitor/images/uploads
scp -r lzm:testing/cpmonitor/images/uploads cpmonitor/images/
```

> To find out, on which migration version this is based use:

Reviewer: ```suggestion
To find out, on which migration version this database is based use:
```

```sh
ssh -tt lzm docker exec -it djangoapp python manage.py showmigrations --settings=config.settings.production-container
```

> Possibly migrate, test this, and check that the size is reasonable. Then make it available to others with:

Reviewer: ```suggestion
Possibly migrate, test the data, and check that the size is reasonable. Then make it available to others with:
```
